server/src/main/java/org/opensearch/rest/action/cat/RestShardsAction.java

```diff
@@ -140,6 +140,7 @@ protected Table getTableWithHeader(final RestRequest request) {
             .addCell("prirep", "alias:p,pr,primaryOrReplica;default:true;desc:primary or replica")
             .addCell("state", "default:true;alias:st;desc:shard state")
             .addCell("docs", "alias:d,dc;text-align:right;desc:number of docs in shard")
```

Comment:
Could we please use `docs.count` and `docs.deleted` as columns (to align with similar segments and indices APIs)? Also, this change could be considered a breaking one for `2.x`, I would suggest to hide it behind the `detailed` request parameter (so the default API response won't change). Please also adapt `RestShardsActionTests`.

Reply:
>Could we please use docs.count and docs.deleted as columns

Have made changes for docs.deleted and kept it as a non-default entry in the API call (will be visible only when added in the headers explicitly) but changing docs to docs.count would be a breaking change - so haven't modified that.

>I would suggest to hide it behind the detailed request parameter

I couldn't find any `detailed` parameter for _cat/shards. Did you mean exposing it only when it is mentioned in the header ? (that has been already done)

>Please also adapt RestShardsActionTests

Since I have kept it as a non-default value and added it in the end of the table, it is not breaking RestShardsActionTests currently.

Reply:
Thanks @rayshrey 

> Have made changes for docs.deleted and kept it as a non-default entry in the API call (will be visible only when added in the headers explicitly) but changing docs to docs.count would be a breaking change - so haven't modified that.

For 3.x we could make a breaking change (not for 2.x)

> I couldn't find any detailed parameter for _cat/shards. Did you mean exposing it only when it is mentioned in the header ? (that has been already done)


```
        boolean detailed = false;
        if (request != null) {
            detailed = Boolean.parseBoolean(request.param("detailed"));
        }
```

Check for example `RestCatSegmentReplicationAction` on how it is used

> Since I have kept it as a non-default value and added it in the end of the table, it is not breaking RestShardsActionTests currently.

It is not about breaking but testing - please make sure the functionality (new column) is covered

Reply:
For the detailed parameter, I think it would be better if we do that change in 3.x as making only the docs.detailed column when detailed is passed doesn't seem quite right when there are >60 non default columns.(If we include all the non default ones under detailed then that would be a breaking change for 2.x)

The 3.x PR for the breaking change I will raise separately (and will include the detailed parameter in that PR)

Regarding the tests - have included tests for the newly added column in `RestShardsActionTests` in the latest revision.

Reply:
Consider adding a new header and by default keep it hidden as suggested and only visible when you request deleted docs via `h=docs_del` as an explicit flag passed in the query params

Reply:
Thanks @rayshrey , I think @Bukhtawar is a good middle ground

Reply:
@reta We are implementing the same thing that @Bukhtawar proposed.
Edited the PR Description to avoid any ambiguity.